jupyter_server_proxy/handlers.py

```diff
@@ -28,14 +28,17 @@ def get(self, *args):
         dest = src._replace(path=src.path + '/')
         self.redirect(urlunparse(dest))
 
-class LocalProxyHandler(WebSocketHandlerMixin, IPythonHandler):
-
+class ProxyHandler(WebSocketHandlerMixin, IPythonHandler):
+    """
+    A tornado request handler that proxies HTTP and websockets from
+    a given host/port combination.
+    """
     def __init__(self, *args, **kwargs):
         self.proxy_base = ''
         self.absolute_url = kwargs.pop('absolute_url', False)
         super().__init__(*args, **kwargs)
 
-    async def open(self, port, proxied_path=''):
+    async def open(self, host, port, proxied_path=''):
```

Comment:
I somehow don't want this to be 'open' here, since that too can be accidentally used directly by users to gain access to arbitrary websocket domains without CORS protection. Can we move this open method to be called proxy_open (to match the 'proxy' method), and call it from 'open' in LocalProxyHandler and friends? This makes it so that each handler that inherits from this *must* implement its own methods for open, http_get, put, post, etc.

Reply:
Makes sense.